src/hotp/mod.rs

```diff
@@ -1,3 +1,85 @@
-pub fn generate() {}
+/// Default layer to generate a HOTP
+/// # Arguments
+///
+/// * `digest` - Byte digest generated by hashing the secret key
+///
+pub fn generate_default(

```

Comment:
I would rather this be generate, and then the _other_ generate be name something else.


most likely, most consumer are not going to want to provide their own hash. IMO the more accessible lib method should be the most convenient to use. 

Other than just naming, im totally down to de couple the impls of digest and otp generation